bchrpc/server.go

```diff
@@ -1544,6 +1601,24 @@ func marshalTransaction(tx *bchutil.Tx, confirmations int32, blockHeader *wire.B
 	return respTx
 }
 
+// setInputMetadata will set the value, previous script, and address for each input in the mempool transaction
+// from blockchain data adjusted upon the contents of the transaction pool.
+// Used when no s.txIndex is available
```

Comment:
Believe this should be WHEN the txIndex is available. The conditions above make sure it is not nil.